Svg/Basic Shapes/SvgImage.cs

```diff
@@ -97,26 +97,37 @@ private void DisposeImage()
 
         public override RectangleF GetBounds()
         {
-            // if a width/height is set explicitly, use that
-                if (!Width.IsNone && !Width.IsEmpty && !Height.IsNone && !Height.IsEmpty)
-                    return RectangleF.Create(Location.ToDeviceValue(null, this),
-                        SizeF.Create(Width.ToDeviceValue(null, UnitRenderingType.Horizontal, this),
-                            Height.ToDeviceValue(null, UnitRenderingType.Vertical, this)));
-
-                var bmp = _img as Image;
-                var svg = _img as SvgFragment;
-
-                if (bmp != null)
-                {
-                    return RectangleF.Create(Location.ToDeviceValue(null, this), SizeF.Create(bmp.Width, bmp.Height));
-                }
-                if (svg != null)
+            if (this.ClipPath != null)
+            {
+                SvgClipPath clipPath = this.OwnerDocument.GetElementById<SvgClipPath>(this.ClipPath.ToString());
```

Comment:
can you maybe use `this.ClipPath.Uri.OriginalString` instead? That way we might save some allocation (ToString always allocates a new string)